code/game/objects/items/weapons/storage/lockbox.dm

```diff
@@ -18,7 +18,9 @@
 	var/tracked_access = "It doesn't look like it's ever been used."
 	health = 50
 
-
+/obj/item/weapon/storage/lockbox/proc/damage()
+	if(istype(src, /obj/item/weapon/gun))
```

Comment:
You are typechecking the wrong thing. src means "the current object", and will always be a lockbox in this case, because the code you've written is for lockboxes.